src/sql/src/plan/statement/ddl.rs

```diff
@@ -3542,14 +3543,54 @@ generate_extracted_config!(
     (EnableLetrecFixpointAnalysis, Option<bool>, Default(None))
 );
 
+/// Convert a [`CreateClusterStatement`] into a [`Plan`].
+///
+/// The reverse of [`unplan_create_cluster`].
 pub fn plan_create_cluster(
+    scx: &StatementContext,
+    stmt: CreateClusterStatement<Aug>,
+) -> Result<Plan, PlanError> {
+    let plan = plan_create_cluster_inner(scx, stmt)?;
+
+    // Roundtrip through unplan and make sure that we end up with the same plan.
+    if let CreateClusterVariant::Managed(_) = &plan.variant {
+        let stmt = unplan_create_cluster(scx, plan.clone())
+            .map_err(|e| PlanError::Replan(e.to_string()))?;
+        let create_sql = stmt.to_ast_string_stable();
+        let stmt = parse::parse(&create_sql)
+            .map_err(|e| PlanError::Replan(e.to_string()))?
+            .into_element()
+            .ast;
+        let (stmt, _resolved_ids) =
+            names::resolve(scx.catalog, stmt).map_err(|e| PlanError::Replan(e.to_string()))?;
+        let stmt = match stmt {
+            Statement::CreateCluster(stmt) => stmt,
+            stmt => {
+                return Err(PlanError::Replan(format!(
+                "replan does not match: plan={plan:?}, create_sql={create_sql:?}, stmt={stmt:?}"
+            )))
+            }
+        };
+        let replan =
+            plan_create_cluster_inner(scx, stmt).map_err(|e| PlanError::Replan(e.to_string()))?;
+        if plan != replan {
+            return Err(PlanError::Replan(format!(
+                "replan does not match: plan={plan:?}, replan={replan:?}"
+            )));
+        }
+    }
```

Comment:
Once we start to have more `unplan`s, this could be generalized to look something like this:

```Rust
fn plan(stmt: Statement<Aug>) -> Result<Plan, PlanError> {
    let plan = plan_inner(stmt)?;
    
    if let Some(stmt) = unplan(plan.clone()) {
        let create_sql = stmt.to_ast_string_stable();
        let stmt = parse::parse(&create_sql)
            .map_err(|e| PlanError::Replan(e.to_string()))?
            .into_element()
            .ast;
       let (stmt, _resolved_ids) =
            names::resolve(scx.catalog, stmt).map_err(|e| PlanError::Replan(e.to_string()))?;
       let replan = plan_inner(stmt).map_err(|e| PlanError::Replan(e.to_string()))?;
       if plan != replan {
            return Err(PlanError::Replan(format!(
                "replan does not match: plan={plan:?}, replan={replan:?}"
            )));
        }
    }

    Ok(plan)
}

fn plan_inner(stmt: Statement<Aud>) -> Result<Plan, PlanError> {
    ...
}

fn unplan(plan: Plan) -> Result<Option<Statement<Aug>>, PlanError> {
    match plan {
        Plan::CreateClusterPlan(plan) => Ok(Some(unplan_create_cluster(plan)?)),
        ...
        // These plans cannot be unplanned.
        _ => Ok(None),
    }
}
```